python-checks/src/main/java/org/sonar/python/checks/NumpyIsNanCheck.java

```diff
@@ -48,17 +57,63 @@ private static void checkForIsNan(SubscriptionContext ctx) {
   }
 
   private static void checkOperand(SubscriptionContext ctx, Expression operand, BinaryExpression be) {
-    HasSymbol cast;
-    if (operand.is(Tree.Kind.QUALIFIED_EXPR)) {
-      cast = (QualifiedExpression) operand;
-    } else if (operand.is(Tree.Kind.NAME)) {
-      cast = (Name) operand;
+    Optional.ofNullable(getSymbolFromExpression(operand))
```

Comment:
since the `symbol` method is declared in the `HasSymbol` interface and both `Name` and `QualifiedExpression` are inherited from it, and the result of the `getSymbolFromExpression` method is used only in `Optional.ofNullable` - I'd suggest removing it and use `TreeUtils.getSymbolFromTree` instead:
```java
TreeUtils.getSymbolFromTree(operand)
 .map(Symbol::fullyQualifiedName)
      .filter("numpy.nan"::equals)
      .ifPresent(fqn -> {
        PreciseIssue issue = ctx.addIssue(be, MESSAGE);
        addQuickFix(issue, operand, be);
      });
```

Reply:
Simplified in e9b3f58c73c70c1e6796be327cff00e07ab95025.